src/app/main/main.component.ts

```diff
@@ -109,9 +114,13 @@ export class MainComponent implements OnInit, OnDestroy {
     this.columnNames = [];
     this.rows = [];
 
+    this.project = this._route.snapshot.paramMap.get("project")
+    this.dataset = this._route.snapshot.paramMap.get("dataset")
+    this.table = this._route.snapshot.paramMap.get("table")
```

Comment:
I'm guessing these URL parameters should be validated somehow. Is there a list of valid characters for projects, datasets, and tables? Otherwise there is some chance of SQL injection here.

Reply:
Yes, without validation this opens GeoViz for SQL ingestion attack. This code should check the fields, project: ASCII letters, digits, and hyphens, and must be between 6 and 30 characters, dataset and tables: letters (upper or lower case), numbers, and underscores, up to 1024. 
I would also consider adding a prompt, asking user, 'do you want to query proj.dataset.table?' - as it might cost them money.

Reply:
Done.